Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,479 @@
+#include "WarpXOpenPMD.H"
+#include "FieldIO.H"  // for getReversedVec
+
+WarpXOpenPMDPlot::WarpXOpenPMDPlot(bool oneFilePerTS,
+                       const std::string& openPMDFileType)
+  :m_Series(nullptr),
+   m_OneFilePerTS(oneFilePerTS),
+   m_OpenPMDFileType(openPMDFileType)
+{}
+
+WarpXOpenPMDPlot::~WarpXOpenPMDPlot()
+{
+  if (nullptr != m_Series) {
+    m_Series->flush();
+    delete m_Series;
+  }
+}
+
+
+//
+//
+//
+void WarpXOpenPMDPlot::GetFileName(std::string& filename, int ts)
+{
+  std::string dir = "diags/";
+  std::string prefix="/plot";
+
+  // Write openPMD format: only for level 0
+  filename = std::string(dir);
+  filename.append(m_OpenPMDFileType);
+
+  if (m_OneFilePerTS)
+    filename += amrex::Concatenate(prefix, ts);
+  else
+    filename.append(prefix);
+
+  filename += std::string(".") + m_OpenPMDFileType;
+}
+
+
+void WarpXOpenPMDPlot::SetOpenPMDType(const std::string& type)
+{
+  m_OpenPMDFileType = type;
+}
+
+void WarpXOpenPMDPlot::SetStep(int ts)
+{
+  if (ts < 0)
+    return;
+
+  m_CurrentStep =  ts;
+
+  std::string filename;
+  if (m_OneFilePerTS) {
+    GetFileName(filename, ts);
+    Init(filename, openPMD::AccessType::CREATE);
```

Comment:
As noted above: this can be simplified without passing `ts`.
Furthermore, let's call `GetFileName` directly in `Init`, so we do not need a temporary variable here.